src/http/client.cr

```diff
@@ -53,6 +53,9 @@
 # of the returned IO (or used for creating a String for the body). Invalid bytes in the given encoding
 # are silently ignored when reading text content.
 class HTTP::Client
+  # The set of possible valid body types
+  alias BodyType = String | Bytes | IO | Nil
```

Comment:
Maybe `BodyType` shouldn't include `Nil`, because it means "no body", then if you want to allow no body then thats just `BodyType?`? I'm 50/50 on this change.